File: src/main/java/net/dontdrinkandroot/gitki/wicket/model/CurrentUserFullNameModel.java
package net.dontdrinkandroot.gitki.wicket.model;

import net.dontdrinkandroot.gitki.model.User;
import net.dontdrinkandroot.gitki.wicket.GitkiWebSession;
import org.apache.wicket.model.IModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class CurrentUserFullNameModel implements IModel<String>
{
    @Override
    public String getObject() {
        User user = GitkiWebSession.get().getUser();
        if (null == user) {
            return "n/a";
        }

        return user.getFullName();
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/model/CurrentUserFullNameModel.kt
package net.dontdrinkandroot.gitki.wicket.model

import net.dontdrinkandroot.gitki.wicket.getCurrentUser
import org.apache.wicket.model.IModel

class CurrentUserFullNameModel : IModel<String> {

    override fun getObject(): String {
        val user = getCurrentUser() ?: return "n/a"
        return user.fullName
    }
}